src/frontend/src/flows/manage.ts

```diff
@@ -150,94 +203,82 @@ export const renderManage = async (
+  // cannot be added again
+  try {
+    await withLoader(() =>
+      connection.add(
+        userNumber,
+        deviceName,
+        { unknown: null },
+        { authentication: null },
+        newDevice.getPublicKey().toDer(),
+        newDevice.rawId
+      )
+    );
+  } catch (error) {
+    await displayFailedToAddTheDevice(error);
+  }
+  renderManage(userNumber, connection);
+};
+
+// Initializes the management page.
 const init = async (
   userNumber: bigint,
   connection: IIConnection,
   devices: DeviceData[]
 ) => {
   // TODO - Check alias for current identity, and populate #nameSpan
   initLogout();
-  const setupRecoveryButton = document.querySelector(
-    "#addRecovery"
-  ) as HTMLButtonElement;
 
-  setupRecoveryButton.onclick = async () => {
-    await setupRecovery(userNumber, connection);
-    renderManage(userNumber, connection);
-  };
+  // Add the buttons for adding devices and recovery mechanism
 
-  const addAdditionalDevice = document.querySelector(
+  // Add device
+  const addAdditionalDeviceButton = document.querySelector(
     "#addAdditionalDevice"
   ) as HTMLButtonElement;
 
-  addAdditionalDevice.onclick = async () => {
-    let newDevice: WebAuthnIdentity;
-    try {
-      newDevice = await WebAuthnIdentity.create({
-        publicKey: creationOptions(devices),
-      });
-    } catch (error) {
-      await displayError({
-        title: "Failed to add new device",
-        message: html`
-          We failed to add your new device.<br />
-          If you're trying to add a device that is not attached to this machine
-          try following the instructions at<br />
-          <a
-            target="_blank"
-            href="https://sdk.dfinity.org/docs/ic-identity-guide/auth-how-to.html#_add_a_device"
-            >https://sdk.dfinity.org/docs/ic-identity-guide/auth-how-to.html#_add_a_device</a
-          >
-        `,
-        detail: error.message,
-        primaryButton: "Back to manage",
-      });
-      return renderManage(userNumber, connection);
-    }
-    const deviceName = await pickDeviceAlias();
-    if (deviceName === null) {
-      return renderManage(userNumber, connection);
-    }
-    // TODO check whether newDevice is already registered,
-    // or better, pass existing devices to `.create` so that they
-    // cannot be added again
-    try {
-      await withLoader(() =>
-        connection.add(
-          userNumber,
-          deviceName,
-          { unknown: null },
-          { authentication: null },
-          newDevice.getPublicKey().toDer(),
-          newDevice.rawId
-        )
-      );
-    } catch (error) {
-      await displayError({
-        title: "Failed to add the new device",
-        message:
-          "We failed to add the new device to this Identity Anchor. Please try again",
-        detail: error.message,
-        primaryButton: "Back to manage",
-      });
-    }
+  addAdditionalDeviceButton.onclick = async () =>
```

Comment:
```suggestion
  addAdditionalDeviceButton.onclick = () =>
```

No need for this to be async, since we're not awaiting in here.